### ISSUE TITLE
"Max KSP version" format needs standardization.

### ISSUE THREAD

**Issue by Prezombie** [KSP-CKAN/NetKAN#5356]
There seems to be a lot of different ways mods have max version labelled, and it screws up sorting.

"1.2" and "1.2.99" do the same thing, enabling use on all 1.2.x, but "1.2.0" only works for exactly that version. Mods which only work for 1.2.0 but not 1.2.1 should be labelled that, not "1.2". It should be standardized to "1.2.x"

Some go even further with "1.99.99" or even "99.99.99". This is getting silly.

A blank entry and "all versions" is identical, and should be merged.

Any entry with a plus sign, like "0.25+" should be re-evaluated. and set to All versions, 1.x.x, or the accurate max version.

**Comment by politas** [comment]
All the unusual choices are there because the mod authors set them that way.

**Comment by politas** [comment]
The biggest problem we have is that Spacedock only allows a single KSP version , and they use the exact version numbers Squad uses. Squad don't put out "1.2.0", they put out "1.2". It's quite a nuisance.

**Comment by ValynEritai** [comment]
I'll admit, the reason who GrindyScience and LessGrindyScience use 9.99.999 is because there's no standard on how to make it available on all versions... or at least from SpaceDock anyway. 😐

**Comment by politas** [comment]
@ValynEritai , we can override that with "Any" in CKAN if that makes sense.

**Comment by HebaruSan** [comment]
> Any entry with a plus sign, like "0.25+" should be re-evaluated. and set to All versions, 1.x.x, or the accurate max version.

Agreed, and this was done in #2131. (That format was used when a mod had only a minimum version defined, which isn't really relevant to the maximum version.)

**Comment by yalov** [comment]
You can add recommendation to wiki about version patterns. So `1.4.x` is better than `1.4.9` ? 

(along with recommendations about mods versioning — `x.y.z`, not `Vx.y.z` or `any_version_because_its_a_MM_patch`. https://semver.org)


There is still plenty nines, so
``` c#
 if      ((major+1) % 10 == 0) str = "Any";
 else if ((minor+1) % 10 == 0) str = major +".x.x";
 else if ((patch+1) % 10 == 0) str = major +"."+ minor +  ".x";
```

**Comment by HebaruSan** [comment]
Sorry, are you reporting a problem, suggesting a change, or something else? The wiki is editable to anyone, that's part of the common notion of what a wiki is.

(The link in my previous comment was was wrong previously, it's updated now to point to the right )

**Comment by yalov** [comment]
> Sorry, are you reporting a problem, suggesting a change, or something else?

I am participating in open issue called: "'Max KSP version' format needs standardization".
I am asking what you choose for that, "1.4.x" and 1.x.x and "Any"?

> we can override that with "Any"
or more intensive override, code above

**Comment by yalov** [comment]
How do you want standardize that? this way? 
 ```
   "KSP_VERSION_MAX": {
        "MAJOR": 1,
        "MINOR": x,
        "PATCH": x
    }
```
```
    "KSP_VERSION_MAX": "Any"
```

**Comment by HebaruSan** [comment]
This issue is talking about how the code formats the column of the "Max KSP Version" _column_ in the GUI, not metadata.

**Comment by yalov** [comment]
then I propose more intensive override, code above

**Comment by yalov** [comment]
> I'll admit, the reason why GrindyScience and LessGrindyScience use 9.99.999 is because there's no standard on how to make it available on all versions

so we need one.

> ### Versions
> We can recommend you to stick to [Semantic Versioning](https://semver.org) (MAJOR.MINOR.PATCH) of your mod.
> 
> For making mod available on all versions you can set 
> "KSP_VERSION_MAX" to "9.9.9", "1.9.9" or "1.4.9" depending from target interval. 
They will be changed to "Any", "1.x.x" and "1.4.x" respectively in the GUI "Max KSP Version" column,

**Comment by HebaruSan** [comment]
What would be the advantage of doing that, rather than showing the value accurately?

**Comment by yalov** [comment]
> Some go even further with "1.99.99" or even "99.99.99". This is getting silly.

So, advantage is ... un-silly-fying

**Comment by HebaruSan** [comment]
How is that silly, if that's what the author of those mods chose as the max version?

**Comment by HebaruSan** [comment]
BTW, the "99.99.99" value is just incorrect, attempting to fix it in KSP-CKAN/CKAN-meta#1338 and Dunbaratu/LaserDist#23.

**Comment by HebaruSan** [comment]
Fixed in KSP-CKAN/CKAN#2437.
